pkg/test/api/api.go

```diff
@@ -0,0 +1,51 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package api
```

Comment:
Should api be simply "test"? The stuff in API would be the main package that the users be interacting with, so it probably needs to have a general, easy-to-understand/guess name (That's why I picked "test", but we can pick another one).

Reply:
I had to make a subpackage because there was a circular dependency between the top-level `test` package and `/cluster`.

Reply:
FYI, we can get rid of the circular dep if we just remove the `Get<Cluster|Local>Environment` helper methods.

Reply:
I think we can get rid of them and come up with a slightly different entry point.
loca.NewEnvironment() and cluster.GetEnvironment() perhaps?

Reply:
Agreed ... I've just removed them entirely for now, since it's more golang-ish to just create the objects in-place. All the `api` structs are moved back to `test`.